### PR TITLE
fix: fix azure portion of cloud uploader

### LINKED ISSUE #7512: fix and rewrite Azure cloud image uploader
1. Correctly propagate errors back, drop `log.Printf()` for errors
2. Drop ARM templates, use native APIs.
3. Correctly handle restarted runs - re-uploading to the same image tag.

### PR DESCRIPTION
**Pull request by ro11net**
Correctly propagate errors back. Drop ARM templates and use native APIs. Correctly handle restarted runs for creating image versions.
- fixes #7512

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

File: api/resource/definitions/perf/perf.proto [hunk 1]
```diff
@@ -81,4 +81,3 @@ message MemorySpec {
   uint64 direct_map2m = 47;
   uint64 direct_map1g = 48;
 }
-
```
Comment (smira) [inline api/resource/definitions/perf/perf.proto:84]: these changes shouldn't be there

Comment (ro11net) [reply api/resource/definitions/perf/perf.proto:84]: I didn't modify any of those files so I'm not sure why they are.

**Review by smira — APPROVED** [review]

**Pull request merged**